src/System.Text.Json/System/Text/Json/JsonParseObject.cs

```diff
@@ -182,6 +182,19 @@ public bool TryGetValue(string propertyName, out JsonObject value)
             }
         }
 
+        public int Length
+        {
+            get
+            {
+                var record = Record;
+                if (record.Type != JsonValueType.Array)
+                {
+                    throw new NullReferenceException();
```

Comment:
Yeah, I just noticed TryGetValue does the same thing. @ahsonkhan, is there a reason for NullRefException here? If not I would appreciate if @Bobris changed TryGetValue too.